zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/usertask/UserTaskProcessor.java

```diff
@@ -192,6 +213,26 @@ private void createTaskListenerJob(
         .ifLeft(failure -> incidentBehavior.createIncident(failure, context));
   }
 
+  private void writeRejectionForCommand(
+      final TypedRecord<UserTaskRecord> command,
+      final UserTaskRecord persistedRecord,
+      final UserTaskIntent intent) {
+    stateWriter.appendFollowUpEvent(persistedRecord.getUserTaskKey(), intent, persistedRecord);
+
+    final var recordRequestMetadata =
+        userTaskState.findRecordRequestMetadata(persistedRecord.getUserTaskKey());
+
+    recordRequestMetadata.ifPresent(
+        metadata -> {
+          responseWriter.writeRejection(
+              command,
+              RejectionType.INVALID_STATE,
+              USER_TASK_COMPLETION_REJECTION.formatted(persistedRecord.getUserTaskKey()),
+              metadata.getRequestId(),
+              metadata.getRequestStreamId());
+        });
```

Comment:
@korthout My idea is that we should send a response right after denying the User Task completion operation to prevent the Tasklist application from experiencing a timeout exception.

However, currently this behaviour can’t be fully tested within Zeebe integration tests, as the related `ZeebeClient` changes will be introduced in a future PR(s) as Ana mentioned [here](https://github.com/camunda/camunda/pull/24320#discussion_r1830968851).
Until we have no option to deny operation using `ZeebeClient` - no one can receive a timeout 😃.

To keep the current PR straightforward, it's enough to write a `COMPLETION_DENIED` event after receiving `DENY_TASK_LISTENER` command.
The response-handling logic could then be addressed in a follow-up PR(s), which Ana is planning as part of task #23709 implementation.